python_modules/dagster/dagster_tests/core_tests/resource_tests/pythonic_resources/test_nesting.py

```diff
@@ -965,3 +965,64 @@ def my_asset(outer: MyBoringOuterResource) -> str:
         "my_asset",
         "SetupTeardownInnerResource yield_for_execution done",
     ]
+
+
+def test_nested_resources_runtime_config_fully_populated() -> None:
+    class CredentialsResource(ConfigurableResource):
+        username: str = "default_username"
+        password: str = "default_password"
+
+    class DBConfigResource(ConfigurableResource):
+        creds: CredentialsResource
+        host: str = "default_host"
+        database: str = "default_database"
+
+    class DBResource(ConfigurableResource):
+        config: DBConfigResource
```

Comment:
nit: I think it would be a bit more clear to make the names reference what we're testing (IE top level resource, nested resource, further nested resource etc